src/fsharp/CheckExpressions.fs

```diff
@@ -5849,12 +5860,28 @@ and TcIteratedLambdas cenv isFirst (env: TcEnv) overallTy takenNames tpenv e =
         let envinner, _, vspecMap = MakeAndPublishSimpleValsForMergedScope cenv env m names
         let byrefs = vspecMap |> Map.map (fun _ v -> isByrefTy cenv.g v.Type, v)
         let envinner = if isMember then envinner else ExitFamilyRegion envinner
+        let vspecs = vs |> List.map (fun nm -> NameMap.find nm vspecMap)
+        
+        // Match up the arginfos with the generated arguments and apply any information extracted from the attributes
+        let envinner =
+            match envinner.eLambdaArgInfos with 
+            | infos :: rest -> 
+                 if infos.Length = vspecs.Length then 
+                    (vspecs, infos) ||> List.iter2 (fun v argInfo -> 
+                        let inlineIfLambda = HasFSharpAttribute cenv.g cenv.g.attrib_InlineIfLambdaAttribute argInfo.Attribs
```

Comment:
I understand the need for `eLambdaArgInfos`, but could we take that info and pass the attributes to the construction of the `Val`, and then do the 'inline' check by looking at the attributes directly from the `Val`.  - but I'm going to assume we actually do not want any attribute information on these Vals.

Reply:
We could post-hoc set attributes on the Val.  I'm not sure of the consequences of that (e.g. would it sudenly activate some attributes like Obsolete on arguments?), so this is enough for this PR

Reply:
Makes sense for this PR.